operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/spec/ManagementSpec.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2024 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.operator.crds.v2alpha1.deployment.spec;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonPropertyDescription;
+import io.sundr.builder.annotations.Buildable;
+import org.keycloak.operator.Constants;
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@Buildable(editableEnabled = false, builderPackage = "io.fabric8.kubernetes.api.builder")
+public class ManagementSpec {
+
+    @JsonPropertyDescription("If separate interface/port should be used for exposing the management endpoints.")
+    private boolean enabled = false; // TODO false for now
```

Comment:
I think we could be opinionated here and not have a first class citizen option for it.

Reply:
Ok, so we should consider having it enabled by default, right? I think it's reasonable, but even when users try to disable it via the `additionalOptions`, the probe attributes must be changed as well. 

But I don't see a big problem here as the management interface should be the default approach for metrics and health exposure. In that case, we could direct them with some workarounds if necessary.

Reply:
> Ok, so we should consider having it enabled by default, right?

+1 for enabling it be default, as it would support the "use safe default" approach. When doing this, we should mention it in the upgrading guide and release notes.